apps/sasquatch/src/androidTest/java/com/microsoft/azure/mobile/sasquatch/activities/AnalyticsTest.java

```diff
@@ -0,0 +1,86 @@
+package com.microsoft.azure.mobile.sasquatch.activities;
+
+
+import android.support.annotation.StringRes;
+import android.support.test.espresso.ViewInteraction;
+import android.support.test.rule.ActivityTestRule;
+import android.view.View;
+
+import com.microsoft.azure.mobile.Constants;
+import com.microsoft.azure.mobile.sasquatch.R;
+
+import org.hamcrest.Matcher;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static android.support.test.espresso.action.ViewActions.replaceText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.RootMatchers.withDecorView;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withChild;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+@SuppressWarnings("unused")
+public class AnalyticsTest {
+
+    @Rule
+    public ActivityTestRule<MainActivity> mActivityTestRule = new ActivityTestRule<>(MainActivity.class);
+
+    @Test
+    public void sendEventTest() throws InterruptedException {
+        onView(allOf(
+                withChild(withText(R.string.title_event)),
+                withChild(withText(R.string.description_event))))
+                .perform(click());
+        onView(withId(R.id.name)).perform(replaceText("test"), closeSoftKeyboard());
+        onView(withText(R.string.send)).perform(click());
+
+        /* Wait trigger interval. */
+        Thread.sleep(Constants.DEFAULT_TRIGGER_INTERVAL + 100);
+
+        /* Check toast. */
+        onToast(withText(R.string.event_before_sending))
+                .check(matches(isDisplayed()));
+
+        /* TODO Wait for sending result. */
```

Comment:
I haven't succeeded with clear implementation "wait to toast" yesterday. Now I wrote more relevant check with `IdlingResource` sync, but anyway have a similar `TODO`.